trojsten/polls/views.py

```diff
@@ -0,0 +1,69 @@
+from django.contrib import messages
+from django.shortcuts import get_object_or_404, redirect, render
+from django.utils import timezone
+
+from .models import Answer, Question, Vote
+
+
+def view_question(request, pk=None):
+    questions = Question.objects.filter(created_date__lte=timezone.now()).order_by("-created_date")
+    if not questions:
+        return render(
+            request, "trojsten/polls/view_question.html", {"questions": [], "user": request.user}
+        )
+    if pk is None:
+        pk = questions[0].pk
+    current = get_object_or_404(Question, pk=pk)
+    current.expired = current.deadline <= timezone.now()
+
+    user = request.user
+    if request.method == "POST":
+        if not user.is_authenticated:
+            # You must be logged in
+            messages.add_message(request, messages.ERROR, "Pre hlasovanie sa musíš prihlásiť.")
+            return redirect("view_question", pk=pk)
+
+        answer_pk = int(request.POST.get("action")[4:])
+        if current.expired:
+            # Deadline is over
+            messages.add_message(
+                request, messages.ERROR, "V tejto ankete už nie je možné hlasovať."
+            )
+            return redirect("view_question", pk=pk)
+        answer = Answer.objects.filter(question=current, pk=answer_pk).first()
+        if answer is None:
+            # Invalid vote
+            messages.add_message(request, messages.ERROR, "Neplatný hlas.")
+            return redirect("view_question", pk=pk)
+
+        existing = Vote.objects.filter(user=user, answer__question=current)
+        if existing:
+            existing[0].answer = answer
+            existing[0].save()
+            for e in existing[1:]:  # This shouldn't happen (user having two votes)

```

Comment:
Toto by malo byt enforcnute databazou (pomocou unique together) a potom to nemusis riesit.

Potom mozes pouzit [update_or_create](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#update-or-create) na zmenu a nemusis sa starat o mazanie.

Reply:
Vies mi prosim povedat, ako pouzit ten unique together (pretoze potrebujem mat unikatnu dvojicu (user, answer->question))?

Reply:
Cez dva podciarkovniky?
(("user", "answer__question"), )

Reply:
Aha, tak v tom pripade by to celkom nefungovalo, ale stale vies urobit ten check v modeli v [Model.validate_unique](https://docs.djangoproject.com/en/3.0/ref/models/instances/#django.db.models.Model.validate_unique)

Reply:
Prepac, no neviem, ako sa ten validate_unique pouziva, neviem najst ziaden rozumny priklad a tych par riadkov dokumentacie mi nepomaha

Reply:
Tu je nejaly example: https://stackoverflow.com/a/55221517

Reply:
Pisem si vlastne testy a validate_unique v nich mi zlyhava = hodi ValidationError, aj ked by nemal. Myslim si, ze je to preto, lebo sa v nom robi
Vote.objects.filter(....).exists(): raise ValidationError
a to vrati True, pretoze ten filter vrati ten objekt samotny. Ma to tak byt?

Reply:
Hej, tak by to malo byt. `validate_unique` by mal raisovat `ValidationError`, pokial objekt s takymi unique fieldami uz v database existuje. tu je naozaj riesenim try/except na `ValidationError` a ohandlovat to tam

Reply:
Pls pushni zmeny, nech vidime kod.